### PR TITLE
129/GUI - Communication

### PR DESCRIPTION
**Pull request by LucasTesnier**
Link the Gui and the server.

Communication in start processus and map size updated.

### CODE REVIEW

**Review by Cavonstavant — APPROVED** [review]
:+1:

**Review by GuyomT — APPROVED** [review]
LGTM

File: src/zappy_gui_src/Core/Core.cpp [hunk 1]
```diff
@@ -81,8 +85,22 @@ void Core::setup(int ac, char **av)
     _getArgs(ac, av);
     str = (char *)_machine.c_str();
     std::cout << "str: " << str << std::endl;
-    if (!c_interface_try_to_connect_to_server(str, std::atoi(_port.c_str())))
+    if (!c_interface_try_to_connect_to_server(str, std::atoi(_port.c_str()))) {
         throw (CoreException("Core setup", "Unable to connect to the server"));
+    }
```
Comment (31Nathan) [inline src/zappy_gui_src/Core/Core.cpp:90]: Useless Bracket

File: src/zappy_gui_src/Core/Core.hpp [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 #include <string>
 #include "SFML.hpp"
+#include "Unpack.hpp"
```
Comment (31Nathan) [inline src/zappy_gui_src/Core/Core.hpp:15]: Missing Space

File: src/zappy_gui_src/Core/Network/Unpack.cpp [hunk 3]
```diff
@@ -90,18 +90,38 @@ Egg Unpack::UnpackEgg(std::vector<std::string> &unpacked)
...
+}
+
 void Unpack::UnpackEntity(Player &p, std::string &packed)
 {
     auto unpacked = stov(packed, '{');
     if (unpacked[0] == "player") {
         p = UnpackPlayer(unpacked);
     } else
-        throw(UnpackException("Unpack" "Invalid packed type player"));
+    throw (std::invalid_argument("Player invalid parsing"));
```
Comment (31Nathan) [inline src/zappy_gui_src/Core/Network/Unpack.cpp:124]: Identation

File: src/zappy_gui_src/Core/Network/Unpack.cpp [hunk 4]
```diff
@@ -110,7 +130,7 @@ void Unpack::UnpackEntity(Tile &t, std::string &packed)
     if (unpacked[0] == "tile") {
         t = UnpackTile(unpacked);
     } else
-        throw(UnpackException("Unpack" "Invalid packed type tile"));
+    throw (std::invalid_argument("Player invalid parsing"));
```
Comment (31Nathan) [inline src/zappy_gui_src/Core/Network/Unpack.cpp:133]: Identation

**Review by 31Nathan — CHANGES_REQUESTED** [review]

**Pull request merged**